js/plugins/cloud-sql-pg/src/index.ts

```diff
@@ -127,26 +146,192 @@ export function configurePostgresRetriever<
  * @param ai A Genkit instance
  * @param params The params for the indexer
  * @param params.tableName The name of the indexer
+ * @param params.engine The engine to use for the indexer
  * @param params.embedder The embedder to use for the retriever
  * @param params.embedderOptions  Options to customize the embedder
- * @returns A Genkit indexer
+ * @param params.metadataColumns The metadata columns to use for the indexer
+ * @param params.idColumn The id column to use for the indexer
+ * @param params.metadataJsonColumn The metadata json column to use for the indexer
+ * @param params.distanceStrategy The distance strategy to use for the indexer
+ * @param params.contentColumn The content column to use for the indexer
+ * @param params.embeddingColumn The embedding column to use for the indexer
+ * @param params.schemaName The schema name to use for the indexer
+ * @param params.chunkSize The chunk size to use for the indexer
+ * @returns Add documents to vector store
  */
 export function configurePostgresIndexer<
   EmbedderCustomOptions extends z.ZodTypeAny,
 >(
   ai: Genkit,
   params: {
     tableName: string;
+    engine: PostgresEngine;
+    schemaName?: string;
+    contentColumn?: string;
+    embeddingColumn?: string;
+    metadataColumns?: string[];
+    ignoreMetadataColumns?: string[];
+    idColumn?: string;
+    metadataJsonColumn?: string;
     embedder: EmbedderArgument<EmbedderCustomOptions>;
     embedderOptions?: z.infer<EmbedderCustomOptions>;
   }
 ) {
+  if (!params.engine) {
+    throw new Error('Engine is required');
+  }
+
+  if (params.metadataColumns && params.ignoreMetadataColumns) {
+    throw new Error('Cannot use both metadataColumns and ignoreMetadataColumns');
+  }
+
+  const {
+    tableName,
+    engine,
+    schemaName,
+    contentColumn,
+    embeddingColumn,
+    metadataColumns,
+    ignoreMetadataColumns,
+    idColumn,
+    metadataJsonColumn,
+    embedder,
+    embedderOptions
+  } = params;
+
+  // Store the final metadata columns at the module level
+  let finalMetadataColumns: string[] = metadataColumns || [];
+
+  async function ensureTableExists() {
+    // Get existing columns and their types if table exists
+    const { rows } = await engine.pool.raw(
+      `SELECT column_name, data_type, is_nullable 
+       FROM information_schema.columns 
+       WHERE table_name = '${tableName}' AND table_schema = '${schemaName || 'public'}'`
```

Comment:
Can we set the default values like https://github.com/averikitsch/genkit/blob/cbef947780b0acac8b08f63e56f2700bcb550a11/js/plugins/cloud-sql-pg/src/index.ts#L137-L139